src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1857,18 +1847,22 @@ private SyntaxToken DiagnosticCreationCheck(List<string> ruleNames, SyntaxToken
                 }
 
                 var locationArgIdentifier = locationArg.Expression as IdentifierNameSyntax;
-                if (locationArgIdentifier == null || locationArgIdentifier.Identifier.Text != locationToken.Text)
+                if (locationArgIdentifier == null)
                 {
                     return emptyResult;
                 }
 
-                return diagnosticToken;
+                SyntaxToken diagnosticToken = GetIdentifierTokenFromLocalDecl(newStatement);
+                emptyResult.Add(diagnosticToken.Text);
```

Comment:
@jepetty You should have a separate list for an actual return, or don't call the list emptyResult. It doesn't make sense that you return emptyResult, but with stuff in it